### ISSUE TITLE
Bug: IR Device Air conditionner no temperature changes

### ISSUE THREAD

**Issue by shyguy81** [OpenWonderLabs/homebridge-switchbot#555]
### Describe The Bug

Temperature doesn't change and still at 24

### To Reproduce

change the temperature to 20 (Celsius)

### Expected behavior

temperature set to 20 (Celsius)

### Relevant log output

```shell
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Set HeatingThresholdTemperature: 24, CurrentTemperatureCached: 24
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin pushAirConditionerStatusChanges Active: 1, disablePushOff: true,  disablePushOn: true
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get HeatingThresholdTemperature: 24
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get TargetHeaterCoolerState: 1, ValidValues: 1,2
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get CurrentTemperature: 24
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get RotationSpeed: 4
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get HeatingThresholdTemperature: 24
[10/22/2022, 10:29:21 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin Get CoolingThresholdTemperature: 24
[10/22/2022, 10:29:23 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin pushAirConditionerDetailsChanges Active: 1, disablePushOff: true,  disablePushOn: true
[10/22/2022, 10:29:23 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin body: {"command":"setAll","parameter":"24,5,1,on","commandType":"command"}
[10/22/2022, 10:29:23 AM] [SwitchBot] [DEBUG] Air Conditioner: Daikin sign: xxxxxxx
[10/22/2022, 10:29:23 AM] [SwitchBot] Air Conditioner: Daikin Sending request to SwitchBot API. body: {"command":"setAll","parameter":"24,5,1,on","commandType":"command"},
```

### Config for homebridge-switchbot

```shell
{
            "name": "SwitchBot",
            "credentials": {
                "token": "xxxx",
                "secret": "xxx",
                "notice": "Keep your token a secret!"
            },
            "options": {
                "irdevices": [
                    {
                        "deviceId": "xxxxxx",
                        "configDeviceName": "xxxxxx",
                        "hide_device": false,
                        "configRemoteType": "Air Conditioner",
                        "connectionType": "OpenAPI",
                        "disablePushOn": true,
                        "disablePushOff": true,
                        "irair": {
                            "hide_automode": true
                        },
                        "external": false,
                        "logging": "standard"
                    },
                    {
                        "deviceId": "xxxxxx",
                        "configDeviceName": "xxxxx",
                        "hide_device": true,
                        "configRemoteType": "TV",
                        "connectionType": "OpenAPI",
                        "external": true,
                        "logging": "standard"
                    },
                    {
                        "deviceId": "xxxxx",
                        "configDeviceName": "xxxxx",
                        "hide_device": true,
                        "configRemoteType": "Speaker",
                        "connectionType": "OpenAPI",
                        "customize": false,
                        "external": true,
                        "logging": "standard"
                    },
                    {
                        "deviceId": "xxxxxxx",
                        "configDeviceName": "xxxxx",
                        "configRemoteType": "Air Conditioner",
                        "connectionType": "OpenAPI",
                        "customize": false,
                        "disablePushOn": true,
                        "disablePushOff": true,
                        "irair": {
                            "hide_automode": true
                        },
                        "external": false,
                        "logging": "standard"
                    },
                    {
                        "deviceId": "xxxxxx",
                        "configDeviceName": "xxxxxx",
                        "configRemoteType": "Air Conditioner",
                        "connectionType": "OpenAPI",
                        "disablePushOn": true,
                        "disablePushOff": true,
                        "irair": {
                            "hide_automode": true
                        },
                        "logging": "standard"
                    }
                ],
                "logging": "standard"
            },
            "platform": "SwitchBot"
        }
```

### Device & Model

NA

### Node.js Version

v14.18.0

### NPM Version

v8.19.2

### Homebridge Version

v1.5.0

### Homebridge Switchbot Plugin Version

v2.2.2

### Operating System

Synology DSM

**Comment by donavanbecker** [comment]
You can turn on the air conditioner right?

**Comment by shyguy81** [comment]
yes i can turn it on

**Comment by taglia** [comment]
Hi, same issue here. I can switch all my AC units on and off, but regardless of what temperature I set in the Home app, the plugin always sends the same command.

Log here, I was trying to change the temperature every time that those commands were sent, but on the home app it just snaps back to 30ºC.

```
[11/2/2022, 6:08:34 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"turnOn","parameter":"default","commandType":"command"},
[11/2/2022, 6:08:36 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"25,1,1,on","commandType":"command"},
[11/2/2022, 6:08:43 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"25,1,1,on","commandType":"command"},
[11/2/2022, 6:08:48 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"25,1,1,on","commandType":"command"},
[11/2/2022, 6:08:49 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"25,1,1,on","commandType":"command"},
[11/2/2022, 6:08:54 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"turnOn","parameter":"default","commandType":"command"},
[11/2/2022, 6:08:55 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"30,2,1,on","commandType":"command"},
[11/2/2022, 6:09:02 PM] [SwitchBot] Air Conditioner: Office Air Conditioner Sending request to SwitchBot API. body: {"command":"setAll","parameter":"30,2,1,on","commandType":"command"},
```

Here is what I see on the Home app.

![CleanShot 2022-11-02 at 18 20 27](https://user-images.githubusercontent.com/612306/199465382-4608ec1e-e065-40b0-a9b9-cb07b512d168.gif)

The SwitchBot app works fine.

**Comment by drlkp** [comment]
Same here I temporary solve fixing scene with fixed temp. Eg Hot, 20° fan speed low. Anyway should be fixed

**Comment by claudiomorabito** [comment]
Same issue here with 2.2.2, actually I'm using 2.3.0-beta.4 which works fine for me. Beta-5 seems to have this feature broken once again.

2.3.0 BETA-4 (working: 24, 21...)
```
23/11/2022, 21:49:47Switchbot BridgeSwitchBotAir Conditioner: Condizionatore salone Sending request to SwitchBot API. body: {"command":"turnOn","parameter":"default","commandType":"command"},
23/11/2022, 21:49:49Switchbot BridgeSwitchBotAir Conditioner: Condizionatore salone Sending request to SwitchBot API. body: {"command":"setAll","parameter":"24,5,1,on","commandType":"command"},
23/11/2022, 21:49:51Switchbot BridgeSwitchBotAir Conditioner: Condizionatore salone Sending request to SwitchBot API. body: {"command":"setAll","parameter":"21,5,1,on","commandType":"command"},

```
2.3.0 BETA-5 (not working: 30, 30, 30...)
```
23/11/2022, 21:29:54Switchbot BridgeSwitchBotAir Conditioner: Condizionatore camera Sending request to SwitchBot API. body: {"command":"turnOn","parameter":"default","commandType":"command"},
23/11/2022, 21:29:56Switchbot BridgeSwitchBotAir Conditioner: Condizionatore camera Sending request to SwitchBot API. body: {"command":"setAll","parameter":"30,5,1,on","commandType":"command"},
23/11/2022, 21:29:58Switchbot BridgeSwitchBotAir Conditioner: Condizionatore camera Sending request to SwitchBot API. body: {"command":"setAll","parameter":"30,5,1,on","commandType":"command"},
23/11/2022, 21:30:02Switchbot BridgeSwitchBotAir Conditioner: Condizionatore camera Sending request to SwitchBot API. body: {"command":"setAll","parameter":"30,5,1,on","commandType":"command"},
```

I hope this will be addressed in the next update.

**Comment by donavanbecker** [comment]
Never changed anything with this in beta5....

**Comment by claudiomorabito** [comment]
so that's a bit weird... tried again beta5 and experienced same behavior while beta4 works... I'm using hoobs, is there something I can specifically check to provide you some more useful data?

**Comment by Shiro-meshi** [comment]
Same Issue Here. Tried v2.3.0 beta6 with no luck...

**Comment by siketyan** [comment]
Same issue here with v2.2.2 and v2.3.0-beta6.
The status updating behaviour seems to result 30 degree celcius always:

```
[11/28/2022, 3:23:56 PM] [SwitchBot] Air Conditioner: Air Conditioner Get TargetHeaterCoolerState: 1, ValidValues: 0,1,2
[11/28/2022, 3:23:56 PM] [SwitchBot] Air Conditioner: Air Conditioner Get CurrentTemperature: 30
[11/28/2022, 3:23:56 PM] [SwitchBot] Air Conditioner: Air Conditioner Get RotationSpeed: 4
[11/28/2022, 3:23:56 PM] [SwitchBot] Air Conditioner: Air Conditioner Get HeatingThresholdTemperature: 30
[11/28/2022, 3:23:56 PM] [SwitchBot] Air Conditioner: Air Conditioner Get CoolingThresholdTemperature: 30
```

**Comment by Kacc36** [comment]
Also having the same issue except it goes to 24 degrees

**Comment by siketyan** [comment]
The temperature problem seems resolved in v2.4.0, but Cooling/HeatingThresholdTemperature is still always 24°C even after setting to 22°C heating. Also mode is always cooling after setting to heat: API request succeeds with heat(5) mode, but the internal state of homebridge-switchbot will preserve in dry(3)

**Comment by jprabawa** [comment]
In case anyone finds this helpful, im on v4.1.5 and this issue still exists for me. The workaround that worked for me:
- Add `"irair": { "hide_automode": true, "pushOn": true }` to the aircon config (not sure if this step is necessary, but adding this in anyway. See https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Configuration for more info
- And then when powering on the AC, change the mode to auto once before changing it back to cool. After this mode change, you will be able to set the temperature without the command sent being always fixed to a constant number
